src/main/java/net/greatstart/controllers/PasswordResetController.java

```diff
@@ -1,117 +1,90 @@
+                                   MailService mailService) {
         this.messages = messages;
         this.userService = userService;
         this.securityService = securityService;
         this.passwordEncoder = passwordEncoder;
-        this.validationService = validationService;
         this.mailService = mailService;
     }
 
     @GetMapping(value = "/user/resetPassword")
-    public ModelAndView showForgotPassForm() {
-        ModelAndView model = new ModelAndView("user/forgotPassword");
-        model.addObject("user", new DtoUser());
-        return model;
-    }
-
-    @PostMapping(value = "/user/resetPassword")
-    public ModelAndView processEmail(HttpServletRequest request, @Valid @ModelAttribute("user") DtoUser dtoUser, Errors
-        errors) {
-        ModelAndView model = new ModelAndView("user/forgotPassword");
-        if (errors.hasErrors()) {
-            return model;
-        }
-        User user = userService.getUserByEmail(dtoUser.getEmail());
+    public ResponseEntity<String[]> processEmail(HttpServletRequest request, @RequestParam("email") String email) {
+        User user = userService.getUserByEmail(email);
         if (user == null) {
-            model.addObject(MESSAGE, messages.getMessage("user.notFound", null, request.getLocale()));
-            return model;
+            throw new ServiceException(messages.getMessage("user.notFound", null, request.getLocale()));
         }
         String passwordResetToken = securityService.createPasswordResetToken(user).getToken();
-        String url = request.getHeader("origin");
+        String url = request.getHeader("referer");
         boolean emailSent = mailService.sendResetTokenEmail(url, request.getLocale(), passwordResetToken, user);
         if (emailSent) {
-            model.addObject(MESSAGE, messages.getMessage("email.sent", null, request.getLocale()));
+            return new ResponseEntity<>(new String[]{messages.getMessage("email.sent", null, request.getLocale())},
+                HttpStatus.OK);
         } else {
-            model.addObject(MESSAGE, messages.getMessage("email.error", null, request.getLocale()));
+            throw new ServiceException(messages.getMessage("email.error", null, request.getLocale()));
         }
-        return model;
     }
 
     @GetMapping(value = "/user/validateToken")
     public ModelAndView validatePassToken(Locale locale, @RequestParam("id") long id, @RequestParam("token") String token) {
         String result = securityService.validatePasswordResetToken(id, token, locale);
         ModelAndView model = new ModelAndView();
         if (result != null) {
-            model.addObject(MESSAGE, messages.getMessage("token.error", null, locale) + result);
-            model.setViewName("login/login");
+            model.setViewName("redirect:/");
+            //TODO: implement displaying errors, maybe in some generic error page
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)